### PR TITLE
Changed many things

### PR DESCRIPTION
**Pull request by StrawWagen**
Added "equip delays" to all emplacements. (when emplacements are spawned, they now have a period of time where they will not fire)
Added system to prevent the use of multiple emplacements at once. to prevent players from spawning 2+ emplacements and firing all of them at once.
Buffed the minigun turret, (turret_bullets) it now does 25 damage ( was 18 ), and reduced spread to 0.015 ( was 0.020 )
Buffed the rifle turret (turret_bullets2) it now does 90 damage ( was 50 )
Added punch to the rifle turret's fire sound.
Buffed the rail cannon turret's projectile (rail_shell), it now spawns another explosion, with a radius of 200, that does 300 damage 
Fixed various common script errors relating to the hit particles of the emplacements.

### CODE REVIEW

**Comment by plally** [comment]
should be using tabs for indentation, to be consistent with this repository

File: lua/entities/rail_shell/init.lua [hunk 1]
```diff
@@ -66,30 +66,31 @@ end
...
+                    util.BlastDamage(inflictor, self.Turret, tr.HitPos, 200, 300)
 
 					local effectdata = EffectData()
 					effectdata:SetOrigin(tr.HitPos)				// Position of Impact
 					effectdata:SetNormal(tr.HitNormal)			// Direction of Impact
 					effectdata:SetStart(self.flightvector:GetNormalized())	// Direction of Round
 					effectdata:SetEntity(self.Entity)			// Who done it?
 					effectdata:SetScale(2.2)				// Size of explosion
-					effectdata:SetRadius(tr.MatType)			// Texture of Impact
+					effectdata:SetRadius(67)			// Texture of Impact
```
Comment (brandonsturgeon) [inline lua/entities/rail_shell/init.lua:93]: Can you assign `67` to a variable name that explains what it is?

File: lua/entities/turret_40mm_frag/init.lua [hunk 2]
```diff
@@ -13,7 +13,7 @@ self.Entity:SetModel( "models/items/ar2_grenade.mdl" )
 self.Entity:PhysicsInit( SOLID_VPHYSICS )      -- Make us work with physics,  	
 self.Entity:SetMoveType( MOVETYPE_NONE )   --after all, gmod is a physics  	
 self.Entity:SetSolid( SOLID_VPHYSICS )        -- CHEESECAKE!    >:3           
-self.Entity:SetColor(255,255,255,255)
+self.Entity:SetColor( Color( 255,255,255,255) )
```
Comment (brandonsturgeon) [inline lua/entities/turret_40mm_frag/init.lua:16]: `Color` objects are technically expensive to create (as are `Vector` and `Angle`s), it's good practice to define them once and use them elsewhere.

File: lua/entities/turret_bullets/shared.lua [hunk 3]
```diff
@@ -14,13 +14,36 @@ ENT.TurretTurnMax=0.7
...
+            
+            -- Setup sounds
+            if SERVER then
+                self:EmitSound( "weapons/ar2/npc_ar2_reload.wav", 70, 50 )
+            end
+            
+        end )
+        
+    end
+end
```
Comment (brandonsturgeon) [inline lua/entities/turret_bullets/shared.lua:33]:
```suggestion
function ENT:EmplacementSetupCheck()
    if self.Setup then return end
    self.Setup = true
        
    timer.Simple( 0.2, function()
        if not IsValid( self ) then return end
        self.LastShot = CurTime() + 5
            
        -- Setup sounds
        if SERVER then
            self:EmitSound( "weapons/ar2/npc_ar2_reload.wav", 70, 50 )
        end
    end )
end

```

File: lua/entities/turret_bullets2/shared.lua [hunk 4]
```diff
@@ -14,13 +14,41 @@ ENT.TurretTurnMax=0.7
...
+                    if not IsValid( self ) then return end
+                    self:EmitSound( "weapons/ar2/npc_ar2_reload.wav", 70, 50 )
+                    
+                end )
+            end
+            
+        end )
+        
+    end
+end
```
Comment (brandonsturgeon) [inline lua/entities/turret_bullets2/shared.lua:38]:
```suggestion
function ENT:EmplacementSetupCheck()
    if self.Setup then return end
    self.Setup = true

    timer.Simple( 0.2, function()
        if not IsValid( self ) then return end
        self.LastShot = CurTime() + 8

        -- Setup sounds
        if SERVER then
            self:EmitSound( "weapons/ar2/ar2_reload.wav", 70, 50 )
            timer.Simple( 3, function()
                if not IsValid( self ) then return end
                self:EmitSound( "weapons/ar2/npc_ar2_reload.wav", 70, 50 )
            end )
        end

    end )
end
```

**Comment by StrawWagen** [comment]
ill just fix it by removing all tab indents

**Comment by plally** [comment]
just use tabs for now please

File: lua/entities/turret_grenade/shared.lua [hunk 5]
```diff
@@ -14,13 +14,41 @@ ENT.TurretTurnMax=0.7
...
+                    if not IsValid( self ) then return end
+                    self:EmitSound( "weapons/ar2/npc_ar2_reload.wav", 70, 50 )
+                    
+                end )
+            end
+            
+        end )
+        
+    end
+end
```
Comment (brandonsturgeon) [inline lua/entities/turret_grenade/shared.lua:38]:
```suggestion
function ENT:EmplacementSetupCheck()
    if self.Setup then return end
    self.Setup = true

    timer.Simple( 0.2, function()
        if not IsValid( self ) then return end
        self.LastShot = CurTime() + 8

        -- Setup sounds
        if SERVER then
            self:EmitSound( "weapons/ar2/ar2_reload.wav", 70, 50 )
            timer.Simple( 3, function()
                if not IsValid( self ) then return end
                self:EmitSound( "weapons/ar2/npc_ar2_reload.wav", 70, 50 )

            end )
        end

    end )
end
```

File: lua/entities/turret_rail/shared.lua [hunk 6]
```diff
@@ -14,12 +14,41 @@ ENT.TurretTurnMax		=0.7
...
+                    if not IsValid( self ) then return end
+                    self:EmitSound( "weapons/ar2/npc_ar2_reload.wav", 70, 50 )
+                    
+                end )
+            end
+            
+        end )
+        
+    end
+end
```
Comment (brandonsturgeon) [inline lua/entities/turret_rail/shared.lua:38]:
```suggestion
function ENT:EmplacementSetupCheck()
    if self.Setup then return end
    self.Setup = true

    timer.Simple( 0.2, function()
        if not IsValid( self ) then return end
        self.LastShot = CurTime() + 8

        -- Setup sounds
        if SERVER then
            self:EmitSound( "weapons/ar2/ar2_reload.wav", 70, 50 )
            timer.Simple( 3, function()
                if not IsValid( self ) then return end
                self:EmitSound( "weapons/ar2/npc_ar2_reload.wav", 70, 50 )
            end )
        end

    end )
end
```

**Review by brandonsturgeon — CHANGES_REQUESTED** [review]
Just some scoping things. Doesn't address the indentation issue, so these should be applied first

File: lua/entities/rail_shell/init.lua (hunk 1 above)
Comment (StrawWagen) [reply lua/entities/rail_shell/init.lua:93]: yes good spot, this is the "fix" for the constant errors that are always spammed by emplacements.
this effect throws an error if its radius isn't the mat type for concrete

**Review by brandonsturgeon — APPROVED** [review]

**Pull request merged**